src/main/java/wooteco/subway/ui/service/StationService.java

```diff
@@ -0,0 +1,40 @@
+package wooteco.subway.ui.service;
+
+import java.util.List;
+
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.stereotype.Service;
+
+import wooteco.subway.dao.StationDao;
+import wooteco.subway.domain.Station;
+import wooteco.subway.dto.request.StationRequest;
+import wooteco.subway.dto.response.StationResponse;
+
+@Service
+public class StationService {
+    private StationDao stationDao;
+
+    public StationService(StationDao stationDao) {
+        this.stationDao = stationDao;
+    }
+
+    public StationResponse create(StationRequest stationRequest) {
+        Station station = new Station(stationRequest.getName());
+        Station newStation;
+        try {
+            newStation = stationDao.save(station);
+            return StationResponse.from(newStation);
+        } catch (DuplicateKeyException ignored) {
+            throw new IllegalStateException("이미 존재하는 역 이름입니다.");
+        }
```

Comment:
이런 경우에는 try 내부에서 모두 관리해줘도 될 것 같아요!! catch에서 throw를 하기 때문에 코드 상에 문제가 발생하진 않을거에요! 코드를 좀 더 가독성이 좋게 관리할 수 있을 것 같아요 :)

```suggestion
        try {
            Station station = new Station(stationRequest.getName());
            Station newStation = stationDao.save(station);
            return StationResponse.from(newStation);
        } catch (DuplicateKeyException ignored) {
            throw new IllegalStateException("이미 존재하는 역 이름입니다.");
        }
```

Reply:
저는 에러가 발생할 수 있는 부분만 try 에 넣으면서 어떤 부분이 예외 상황을 야기하는지 명시적으로 나타낼 수 있다고 생각해서 저런 방식으로 작성했던건데, 기존 코드의 23번째 줄은 try 안으로 넣어도 될 것 같다는 생각이 들었어요!
그런데 new Station을 하는 기존 코드의 22번째 줄도 try 안에 넣으신 이유가 있나용??

Reply:
생성한 Station이 catch 로직에서 사용된다고 한다면 어쩔 수 없이 try 문 밖에 둬야 하긴 해요. 그런데 지금 상황에서는 그럴 이유가 없다보니 try 문 내에 두는게 코드 가독성도 더 좋고 읽기 좋다고 생각해서 그렇게 제안드려봤습니다!! 결국 우리가 행하려는 로직이 모두 try 문 내에 있는 것이 어색하지 않은 것 같다고 생각해서요!!

Reply:
넵 좋은 의견 감사합니다 :)